spec/requests/openid_google_provider_callback_spec.rb

```diff
@@ -0,0 +1,87 @@
+#++
+
+require 'spec_helper'
+require 'rack/test'
+
+describe 'OpenID Google provider callback' do
+  include Rack::Test::Methods
+  include API::V3::Utilities::PathHelper
+
+  it 'redirects user without errors', webmock: true, with_settings: {
+    plugin_openproject_openid_connect: {
+      "providers" => { "google" => { "identifier" => "identifier", "secret" => "secret" } }
+    }
+  } do
+    with_enterprise_token :openid_providers
+    auth_hash = { "state" => "623960f1b4f1020941387659f022497f536ad3c95fa7e53b0f03bdbf36debd59f76320801ea2723df520",
+                  "code" => "4/0AVHEtk6HMPLH08Uw8OVoSaAbd2oTi7Z6wOlBsMQ99Yj3qgKhhyKAxUQBvQ2MZuRzvueOgQ",
+                  "scope" => "email profile https://www.googleapis.com/auth/userinfo.email openid https://www.googleapis.com/auth/userinfo.profile",
+                  "authuser" => "0",
+                  "prompt" => "none" }
+    uri = URI('/auth/google/callback')
+    uri.query = URI.encode_www_form([['code', auth_hash['code']],
+                                     ['state', auth_hash['state']],
+                                     ['scope', auth_hash['scope']],
+                                     ['authuser', auth_hash['authuser']],
+                                     ['prompt', auth_hash['prompt']]])
+    stub_request(:post, "https://accounts.google.com/o/oauth2/token").to_return(
+      status: 200,
+      body: {
+        "access_token" =>
+        "ya29.a0Ael9sCPGoZQiKuMHHVKiaiWV9NatII8T7ZY6XiwTcY-VtvSnmPH53BXDoWGU7OpFY7ctZjY0Qf-Cd_5HHULGoF_m-3WEgMvuO7F11nbYI7qoe95enqneFgDh__vvTxGRAGPpl_Xf7qbXVznh35-DHuvhyPAZmMwaCgYKAQISARASFQF4udJhMeehVtS01I8wd8HL6ReQDw0166",
+        "expires_in" => 3594,
+        "scope" =>
+        "https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/userinfo.profile openid",
+        "token_type" => "Bearer",
+        "id_token" =>
+        "eyJhbGciOiJSUzI1NiIsImtpZCI6IjFhYWU4ZDdjOTIwNThiNWVlYTQ1Njg5NWJmODkwODQ1NzFlMzA2ZjMiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiNDI3NzUwNzQ4MTg2LWQ4OGozamNlYmN2bGlxMmd0a3RiZm1oc2lhNjYxZDU4LmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiNDI3NzUwNzQ4MTg2LWQ4OGozamNlYmN2bGlxMmd0a3RiZm1oc2lhNjYxZDU4LmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTA3NDAzNTExMDM3OTIxMzU1MzA3IiwiZW1haWwiOiJiYTFhc2hwYXNoQGdtYWlsLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJhdF9oYXNoIjoiVFBtc0ZHRng4cjdrb3RiZkJud0xVdyIsImlhdCI6MTY4MDYxMjE5NCwiZXhwIjoxNjgwNjE1Nzk0fQ.IDKlHDVg1d7tAqb8eRiq90T52xnwVX9huDjpdLoJpqr4xlnTrFCdalxJBBHd9Cv39g2KPuJaCU21B59yNAyJP6bl5P8e9Ky-y8wOFcgHqcG5qXcNtxCS3imASCchRTtre8yp9AQGYkTIC0Jh6lWg0trdfO-_idKBsd5naJeaeYdeZGkpQ8D4dxn_odla67BO3y2mUtyE4gEbzyq6wTXDATN4ucM4Dyp3Wdk7YpYYuFN1g-sF6NFl4YqugQ4zk-pYYtPLlPgGiqi3_hO9kYbRDhNBtfbMx568m-CyM2tiOIkb4utPR20scSiRqnY2oxOcd5g9znvJOjtanHM3KVdj5g"
+      }.to_json,
+      headers: { "content-type" => "application/json; charset=utf-8" }
+    )
+    stub_request(:get, "https://www.googleapis.com/oauth2/v3/userinfo?alt=json").to_return(
```

Comment:
I had to remove the `?alt=json` for this to work on my local machine. For the stub it shouldn't make a difference, or does it? Got this error:

```
 1) OpenID Google provider callback redirects user without errors
     Failure/Error: response = get uri.to_s

     WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: GET https://www.googleapis.com/oauth2/v3/userinfo with headers {'Accept'=>'*/*', 'Authorization'=>'Bearer ya29.a0Ael9sCPGoZQiKuMHHVKiaiWV9NatII8T7ZY6XiwTcY-VtvSnmPH53BXDoWGU7OpFY7ctZjY0Qf-Cd_5HHULGoF_m-3WEgMvuO7F11nbYI7qoe95enqneFgDh__vvTxGRAGPpl_Xf7qbXVznh35-DHuvhyPAZmMwaCgYKAQISARASFQF4udJhMeehVtS01I8wd8HL6ReQDw0166', 'Date'=>'Tue, 11 Apr 2023 09:37:58 GMT', 'User-Agent'=>'OpenIDConnect::AccessToken (1.21.3) (2.8.3, ruby 3.2.1 (2023-02-08))'}

       You can stub this request with the following snippet:

       stub_request(:get, "https://www.googleapis.com/oauth2/v3/userinfo").
         with(
         ...
```

Reply:
I receive the opposite after removing :)
I will have a look how to deal with both cases.
![image](https://user-images.githubusercontent.com/11717656/231163594-b8d4a637-63f3-477a-9d8b-9020a6b7d68f.png)